presets/tuning/text-generation/dataset.py

```diff
@@ -0,0 +1,154 @@
+        filename_lower = os.path.basename(file_path).lower()
+        for ext in SUPPORTED_EXTENSIONS:
+            if ext in filename_lower:
+                return ext
+        _, file_ext = os.path.splitext(file_path)
+        return file_ext[1:]  # Remove leading "."
+
+    def shuffle_dataset(self, seed=None):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        self.dataset = self.dataset.shuffle(seed=seed)
+
+    def split_dataset(self):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        assert 0 < self.config.train_test_split <= 1, "Train/Test split needs to be between 0 and 1"
+        if self.config.train_test_split < 1:
+            split_dataset = self.dataset.train_test_split(
+                test_size=1-self.config.train_test_split,
+                seed=self.config.shuffle_seed
+            )
+            return split_dataset['train'], split_dataset['test']
+        else:
+            return self.dataset, None
+
+    def get_dataset(self):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        return self.dataset
+
+    def format_and_preprocess(self):
+        # OAI Compliant: https://platform.openai.com/docs/guides/fine-tuning/preparing-your-dataset
+        # https://github.com/huggingface/trl/blob/main/trl/extras/dataset_formatting.py
+        # https://huggingface.co/docs/trl/en/sft_trainer#dataset-format-support
+        if self.config.messages_column:
+            self.format_conversational()
+        elif self.config.context_column and self.config.response_column:
+            self.format_instruct()
+        elif self.config.response_column:
+            self.dataset = self.dataset.map(
+                lambda example: self.tokenizer(example[self.config.response_column], **self.tokenizer_params),
+                batched=True
+            )
+            self.format_text()
+            return self.config.response_column
+        return None
+
+    def format_text(self):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        if self.config.response_column not in self.dataset.column_names:
+            raise ValueError(f"Column '{self.config.response_column}' does not exist in the dataset. Available columns: {self.dataset.column_names}")
+        self.dataset = self.dataset.select_columns([self.config.response_column])
+
+    def format_instruct(self):
+        """Ensure dataset is formatted for instruct fine tuning"""
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")        
+        required_columns = [self.config.context_column, self.config.response_column]
+        for column in required_columns:
+            if column not in self.dataset.column_names:
+                raise ValueError(f"Column '{column}' does not exist in the dataset. Available columns: {self.dataset.column_names}")
```

Comment:
The logic in line 93-96 appears 3 times, can we make it a separate helper method?

Reply:
made two seperate helpers for this - thanks